pkg/condition/condition.go

```diff
@@ -54,6 +75,19 @@ func IsFalseWithReason(conditions []toolchainv1alpha1.Condition, conditionType t
 	return found && c.Status == apiv1.ConditionFalse && c.Reason == reason
 }
 
+// Count counts the conditions that match the given type/status/reason
+func Count(conditions []toolchainv1alpha1.Condition, conditionType toolchainv1alpha1.ConditionType, status apiv1.ConditionStatus, reason string) int {
+	count := 0
+	for _, c := range conditions {
+		if c.Type == toolchainv1alpha1.TemplateUpdateRequestComplete &&
```

Comment:
```suggestion
		if c.Type == conditionType &&
```
I'm assuming we want to check against `conditionType` here

Reply:
good catch, thanks @rajivnathan!